src/bit-systems/scene-loading.ts

```diff
@@ -108,12 +99,26 @@ function* loadScene(
       }
     });
 
+    // Find and load a nav-mesh
+    let navMesh;
+    const navMeshEid = findChildWithComponent(world, NavMesh, scene);
+    if (navMeshEid) {
+      const navMeshObj = world.eid2obj.get(navMeshEid);
+      // TODO the "as any" here is because of incorrect type definition for getObjectByProperty. It was fixed in r145
+      navMesh = navMeshObj?.getObjectByProperty("isMesh", true as any) as Mesh;
+      // Some older scenes have the nav-mesh component as an ancestor of the mesh itself
+      if (navMesh !== navMeshObj) {
+        console.warn("The `nav-mesh` component should be placed directly on a mesh.");
+      }
+      sceneEl.systems.nav.loadMesh(navMesh, "character");
+    }
+
+    // Create scene physics
     if (findChildWithComponent(world, TrimeshTag, scene) || findChildWithComponent(world, HeightFieldTag, scene)) {
       console.log("heightfield or trimesh found on scene");
     } else {
-      let navMeshEid = findChildWithComponent(world, NavMesh, scene);
-      if (navMeshEid) {
-        inflatePhysicsShape(world, navMeshEid, {
+      if (isHighDensity && navMesh) {
+        inflatePhysicsShape(world, navMesh.eid!, {
```

Comment:
nit:

Can't `navMeshEid` be used here instead of `navMesh` and `navMesh.eid!`?

```typescript
      if (isHighDensity && navMeshEid) {
        inflatePhysicsShape(world, navMeshEid, {
```

If it can be, `navMesh` can be scoped into the `if` block above.

```typescript
    if (navMeshEid) {
      const navMeshObj = world.eid2obj.get(navMeshEid);
      // TODO the "as any" here is because of incorrect type definition for getObjectByProperty. It was fixed in r145
      const navMesh = navMeshObj?.getObjectByProperty("isMesh", true as any) as Mesh;
      // Some older scenes have the nav-mesh component as an ancestor of the mesh itself
      if (navMesh !== navMeshObj) {
        console.warn("The `nav-mesh` component should be placed directly on a mesh.");
      }
      sceneEl.systems.nav.loadMesh(navMesh, "character");
    }
```